app/views/govuk_publishing_components/components/_table.html.erb

```diff
@@ -2,46 +2,63 @@
   caption ||= nil
   head ||= []
   rows ||= []
-  first_cell_is_header ||=false
+  first_cell_is_header ||= false
   caption_classes ||= nil
-  sortable ||=false
+  sortable ||= false
+  filterable ||= false
 %>
 
-<%= GovukPublishingComponents::AppHelpers::TableHelper.helper(self, caption, {
-    sortable: sortable,
-    caption_classes: caption_classes
-  }) do |t| %>
+<div <% if filterable %> data-module="table-filter" <% end %> >
+  <% if filterable %>
+    <div class="govuk-table-filter">
```

Comment:
These changes will need to be tested in the [table spec file](https://github.com/alphagov/govuk_publishing_components/blob/main/spec/components/table_spec.rb) as well. You'd probably want to check that if the filter option is turned on that the input is rendered, and maybe test that the 'no results' text is rendered as well, maybe that the div has the right data-module on it, etc.

Reply:
I have added a new test to the RSpec file for this component. My knowledge of RSpec is minimal though so this will need a more experienced look.